keith-interactive/src/browser/layered/layered-interactive-view.tsx

```diff
@@ -102,106 +107,127 @@ export function renderPositions(current: number, nodes: KNode[], layers: Layer[]
         }
     }
     // position of selected node
-    let curPos = getPositionInLayer(layerNodes, target)
+    let curPos = getPositionInLayer(layerNodes, target, direction)
+
+    // determine reative constraint
+    let cons = undefined
+    if (relCons) {
+        cons = determineCons(nodes, layers, target)
+    }
 
     layerNodes.sort((a, b) => a.properties.positionId - b.properties.positionId)
     if (layerNodes.length > 0) {
         let result = <g></g>
-        // mid of the current layer
-
         let shift = 1
         let x = 0, y = 0;
         // calculate positions between nodes
         for (let i = 0; i < layerNodes.length - 1; i++) {
+            // cons is undefined if target is an adjacent node. If this is the case, the circle should not be filled
+            let fill = cons !== undefined ? cons.relCons !== RelCons.UNDEFINED && curPos === i + shift : curPos === i + shift
             let node = layerNodes[i]
-            // at the old position of the selected node should not be a circle
-            if (!node.selected && !layerNodes[i + 1].selected) {
+            // coordinates for both inspected nodes
+            let nodeY = node.position.y
+            let nodeX = node.position.x
+            let nextNodeY = layerNodes[i + 1].position.y
+            let nextNodeX = layerNodes[i + 1].position.x
+            if (node.selected) {
+                nodeY = node.shadowY
+                nodeX = node.shadowX
+                shift = 0
+                fill = cons !== undefined && cons.node.id === layerNodes[i + 1].id && cons.relCons === RelCons.IN_LAYER_PRED_OF
+            } else if (layerNodes[i + 1].selected) {
+                nextNodeY = layerNodes[i + 1].shadowY
+                nextNodeX = layerNodes[i + 1].shadowX
+                fill = cons !== undefined && cons.node.id === node.id && cons.relCons === RelCons.IN_LAYER_SUCC_OF
+            }
```

Comment:
@Drakae why is fill even calculated at first? Is will always be overridden by the conditional.